src/category_theory/abelian/basic.lean

```diff
@@ -109,6 +117,26 @@ open category_theory
 namespace category_theory.abelian
 variables {C : Type u} [category.{v} C] [abelian C]
 
+/-- An abelian category has finite biproducts. -/
+def has_finite_biproducts : has_finite_biproducts.{v} C :=
+limits.has_finite_biproducts.of_has_finite_products
+
+section to_non_preadditive_abelian
+
+local attribute [instance] has_finite_biproducts
+
+@[priority 100] instance nonpreadditive_abelian : non_preadditive_abelian C :=
+{ has_zero_object := infer_instance,
+  has_zero_morphisms := infer_instance,
+  has_kernels := infer_instance,
+  has_cokernels := infer_instance,
+  has_finite_products := infer_instance,
+  has_finite_coproducts := infer_instance,
+  normal_mono := λ X Y, normal_mono,
+  normal_epi := λ X Y, normal_epi }
```

Comment:
```suggestion
@[priority 100] instance nonpreadditive_abelian : non_preadditive_abelian C := { ..‹abelian C› }
```

Reply:
Only `normal_mono` and `normal_epi` are being copied from the `..`; everything else is done by inference, I think.